<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>670</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1200</width>
    <height>670</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1200</width>
    <height>670</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QLabel" name="OriImage">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>110</y>
      <width>512</width>
      <height>512</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>512</width>
      <height>512</height>
     </size>
    </property>
    <property name="maximumSize">
     <size>
      <width>512</width>
      <height>512</height>
     </size>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="NewImage">
    <property name="geometry">
     <rect>
      <x>700</x>
      <y>110</y>
      <width>512</width>
      <height>512</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>512</width>
      <height>512</height>
     </size>
    </property>
    <property name="maximumSize">
     <size>
      <width>512</width>
      <height>512</height>
     </size>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="OriLabel">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>590</y>
      <width>72</width>
      <height>15</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="NewLabel">
    <property name="geometry">
     <rect>
      <x>960</x>
      <y>600</y>
      <width>72</width>
      <height>15</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton">
    <property name="geometry">
     <rect>
      <x>600</x>
      <y>570</y>
      <width>93</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>PushButton</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit">
    <property name="geometry">
     <rect>
      <x>480</x>
      <y>570</y>
      <width>104</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1200</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="File">
    <property name="title">
     <string>文件</string>
    </property>
    <addaction name="OpenImage"/>
    <addaction name="RecentImage"/>
    <addaction name="separator"/>
    <addaction name="CompressImage"/>
    <addaction name="DecompressImage"/>
    <addaction name="separator"/>
    <addaction name="SaveChange"/>
    <addaction name="SaveImageAs"/>
    <addaction name="ExportStatisticalResult"/>
    <addaction name="separator"/>
    <addaction name="Exit"/>
   </widget>
   <widget class="QMenu" name="Edit">
    <property name="title">
     <string>编辑</string>
    </property>
    <widget class="QMenu" name="menu_Shift_P">
     <property name="title">
      <string>点运算变换</string>
     </property>
     <addaction name="PointFunction1"/>
     <addaction name="PointFunction2"/>
     <addaction name="PointFunction3"/>
     <addaction name="separator"/>
     <addaction name="PointFunction4"/>
     <addaction name="PointFunction5"/>
    </widget>
    <widget class="QMenu" name="menu">
     <property name="title">
      <string>直方图均衡</string>
     </property>
     <addaction name="Equalization_Traditional"/>
     <addaction name="Equalization_Improved"/>
    </widget>
    <widget class="QMenu" name="menu_2">
     <property name="title">
      <string>图像平滑</string>
     </property>
     <addaction name="SmoothingImage_Overlimit"/>
     <addaction name="SmoothingImage_MedianFilter"/>
     <addaction name="SmoothingImage_KNNFilter"/>
     <addaction name="SmoothingImage_ULMFilter"/>
    </widget>
    <addaction name="SetSamplingRate"/>
    <addaction name="SetQuantitativeLevel"/>
    <addaction name="separator"/>
    <addaction name="menu_Shift_P"/>
    <addaction name="menu"/>
    <addaction name="GrayStatistics"/>
    <addaction name="separator"/>
    <addaction name="ImageTranssharp"/>
    <addaction name="separator"/>
    <addaction name="CreateBinaryImage"/>
    <addaction name="TranslateBmp_to_Txt"/>
    <addaction name="separator"/>
    <addaction name="menu_2"/>
    <addaction name="SharpeningImage"/>
    <addaction name="EdgeClose"/>
    <addaction name="HoughTransform"/>
    <addaction name="separator"/>
    <addaction name="OptionalConvolution"/>
   </widget>
   <widget class="QMenu" name="Views">
    <property name="title">
     <string>视图</string>
    </property>
    <addaction name="Views_OpenImage"/>
    <addaction name="Views_SaveImageAs"/>
    <addaction name="separator"/>
    <addaction name="Views_SetSamplingRate"/>
    <addaction name="Views_SetQuantitativeLevel"/>
   </widget>
   <widget class="QMenu" name="menu_3">
    <property name="title">
     <string>特殊</string>
    </property>
    <addaction name="RGB_to_256Color"/>
   </widget>
   <addaction name="File"/>
   <addaction name="Edit"/>
   <addaction name="Views"/>
   <addaction name="menu_3"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="OpenImage">
   <property name="text">
    <string>打开图片</string>
   </property>
   <property name="toolTip">
    <string>打开图片</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="RecentImage">
   <property name="text">
    <string>最近使用的图片</string>
   </property>
  </action>
  <action name="SaveChange">
   <property name="text">
    <string>保存修改</string>
   </property>
   <property name="toolTip">
    <string>保存修改</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="SaveImageAs">
   <property name="text">
    <string>将图片另存为…</string>
   </property>
   <property name="toolTip">
    <string>将图片另存为…</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="ExportStatisticalResult">
   <property name="text">
    <string>导出统计结果</string>
   </property>
   <property name="toolTip">
    <string>导出统计结果</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+E</string>
   </property>
  </action>
  <action name="Exit">
   <property name="text">
    <string>退出</string>
   </property>
   <property name="shortcut">
    <string>Esc</string>
   </property>
  </action>
  <action name="SetSamplingRate">
   <property name="text">
    <string>设置采样率</string>
   </property>
   <property name="toolTip">
    <string>设置采样率</string>
   </property>
   <property name="shortcut">
    <string>Shift+S</string>
   </property>
  </action>
  <action name="SetQuantitativeLevel">
   <property name="text">
    <string>设置量化等级</string>
   </property>
   <property name="toolTip">
    <string>设置量化等级</string>
   </property>
   <property name="shortcut">
    <string>Shift+Q</string>
   </property>
  </action>
  <action name="CreateBinaryImage">
   <property name="text">
    <string>生成位平面二值图</string>
   </property>
   <property name="toolTip">
    <string>生成位平面二值图</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+B</string>
   </property>
  </action>
  <action name="TranslateBmp_to_Txt">
   <property name="text">
    <string>图像转文字</string>
   </property>
   <property name="toolTip">
    <string>图像转文字</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+T</string>
   </property>
  </action>
  <action name="GrayStatistics">
   <property name="text">
    <string>灰度统计</string>
   </property>
   <property name="toolTip">
    <string>灰度统计</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+G</string>
   </property>
  </action>
  <action name="Views_OpenImage">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>打开图片</string>
   </property>
   <property name="toolTip">
    <string>视图_打开图片</string>
   </property>
   <property name="shortcut">
    <string>Alt+O</string>
   </property>
  </action>
  <action name="Views_SaveImageAs">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>将图片另存为…</string>
   </property>
   <property name="toolTip">
    <string>视图_将图片另存为…</string>
   </property>
   <property name="shortcut">
    <string>Alt+S</string>
   </property>
  </action>
  <action name="Views_SetSamplingRate">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>设置采样率</string>
   </property>
   <property name="toolTip">
    <string>视图_设置采样率</string>
   </property>
   <property name="shortcut">
    <string>Alt+Shift+S</string>
   </property>
  </action>
  <action name="Views_SetQuantitativeLevel">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>设置量化等级</string>
   </property>
   <property name="toolTip">
    <string>视图_设置量化等级</string>
   </property>
   <property name="shortcut">
    <string>Alt+Shift+Q</string>
   </property>
  </action>
  <action name="PointFunction1">
   <property name="text">
    <string>函数1：f(x) = x + 50</string>
   </property>
   <property name="toolTip">
    <string>函数1：f(x) = x + 50</string>
   </property>
   <property name="shortcut">
    <string>Shift+F1</string>
   </property>
  </action>
  <action name="PointFunction2">
   <property name="text">
    <string>函数2：f(x) = 1.5x</string>
   </property>
   <property name="toolTip">
    <string>函数2：f(x) = 1.5x</string>
   </property>
   <property name="shortcut">
    <string>Shift+F2</string>
   </property>
  </action>
  <action name="PointFunction3">
   <property name="text">
    <string>函数3：f(x) = -x + 256</string>
   </property>
   <property name="toolTip">
    <string>函数3：f(x) = -x + 256</string>
   </property>
   <property name="shortcut">
    <string>Shift+F3</string>
   </property>
  </action>
  <action name="PointFunction4">
   <property name="text">
    <string>函数4：f(x) = x^2</string>
   </property>
   <property name="toolTip">
    <string>函数4：f(x) = x^2</string>
   </property>
   <property name="shortcut">
    <string>Shift+F4</string>
   </property>
  </action>
  <action name="PointFunction5">
   <property name="text">
    <string>函数5：f(x) = 256 * sin(pi * x / 256)</string>
   </property>
   <property name="toolTip">
    <string>函数5：f(x) = 256 * sin(pi * x / 256)</string>
   </property>
   <property name="shortcut">
    <string>Shift+F5</string>
   </property>
  </action>
  <action name="Equalization_Traditional">
   <property name="text">
    <string>传统方法</string>
   </property>
   <property name="shortcut">
    <string>Shift+T</string>
   </property>
  </action>
  <action name="Equalization_Improved">
   <property name="text">
    <string>改进方法</string>
   </property>
   <property name="shortcut">
    <string>Shift+I</string>
   </property>
  </action>
  <action name="ImageTranssharp">
   <property name="text">
    <string>空间变换</string>
   </property>
   <property name="toolTip">
    <string>空间变换</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+I</string>
   </property>
  </action>
  <action name="SmoothingImage_Overlimit">
   <property name="text">
    <string>超限像素平滑法</string>
   </property>
   <property name="shortcut">
    <string>Shift+O</string>
   </property>
  </action>
  <action name="SmoothingImage_MedianFilter">
   <property name="text">
    <string>中值滤波法</string>
   </property>
   <property name="shortcut">
    <string>Shift+M</string>
   </property>
  </action>
  <action name="SmoothingImage_KNNFilter">
   <property name="text">
    <string>K近邻均值滤波器法</string>
   </property>
   <property name="shortcut">
    <string>Shift+K</string>
   </property>
  </action>
  <action name="SmoothingImage_ULMFilter">
   <property name="text">
    <string>最小均方差滤波器法</string>
   </property>
   <property name="shortcut">
    <string>Shift+U</string>
   </property>
  </action>
  <action name="SharpeningImage">
   <property name="text">
    <string>图像锐化</string>
   </property>
   <property name="shortcut">
    <string>Shift+R</string>
   </property>
  </action>
  <action name="OptionalConvolution">
   <property name="text">
    <string>自由卷积变换</string>
   </property>
   <property name="toolTip">
    <string>自由卷积变换</string>
   </property>
   <property name="shortcut">
    <string>Shift+F</string>
   </property>
  </action>
  <action name="RGB_to_256Color">
   <property name="text">
    <string>真彩图转256色图</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Alt+Shift+R</string>
   </property>
  </action>
  <action name="EdgeClose">
   <property name="text">
    <string>边缘点闭合</string>
   </property>
   <property name="shortcut">
    <string>Shift+E</string>
   </property>
  </action>
  <action name="HoughTransform">
   <property name="text">
    <string>霍夫变换</string>
   </property>
   <property name="shortcut">
    <string>Shift+H</string>
   </property>
  </action>
  <action name="CompressImage">
   <property name="text">
    <string>图像压缩</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="DecompressImage">
   <property name="text">
    <string>图像解压</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+D</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
